tests/list-test.cc

```diff
@@ -0,0 +1,211 @@
+// Copyright Microsoft and CHERIoT Contributors.
+// SPDX-License-Identifier: MIT
+
+#define TEST_NAME "List"
+#include "tests.hh"
+#include <ds/linked_list.h>
+
+namespace
+{
+	using ObjectRing = ds::linked_list::cell::Pointer;
+
+	/**
+	 * Example class we want to link into a doubly linked list.
+	 *
+	 * The class contains a single integer for the purposes of the test.
+	 *
+	 * `ds::linked_list` is an intrusive list: we embed the list node into
+	 * the class we want to link.  There are various implementations of the
+	 * list nodes. Here we use the most simple one
+	 * (`ds::linked_list::cell::Pointer`) which relies on two pointers
+	 * `next` and `prev`.
```

Comment:
This bit of the comment might be more naturally associated with the `using ObjectRing` above?  That `using` could, I think, move into the `LinkedObject` class were that to be a more natural place for it (the `Sentinel<ObjectRing>` below would have to become `Sentinel<LinkedObject::ObjectRing>`, but otherwise nothing changes, I think).

Reply:
Agree, makes sense!